doc/book/box/indexes.rst

```diff
@@ -28,49 +27,225 @@ Indexes have certain limitations. See details on page
 Creating an index
 --------------------------------------------------------------------------------
 
-An index definition may include identifiers of tuple fields and their expected
-**types**. See allowed indexed field types in section
-:ref:`Details about indexed field types <details_about_index_field_types>`.
+It is mandatory to create an index for a space before trying to insert
+tuples into it, or select tuples from it.
 
-..  NOTE::
+The simple :doc:`index-creation </reference/reference_lua/box_space/create_index>`
+operation is:
 
-    A recommended design pattern for a data model is to base primary keys on the
-    first fields of a tuple, because this speeds up tuple comparison.
+..  cssclass:: highlight
+..  parsed-literal::
 
-Let's look at an example where we first define the primary index (named 'primary')
-based on field #1 of each tuple:
+    :extsamp:`box.space.{**{space-name}**}:create_index('{*{index-name}*}')`
 
-..  code-block:: tarantoolsession
+This creates a unique :ref:`TREE <indexes-tree>` index on the first field
+of all tuples (often called "Field#1"), which is assumed to be numeric.
 
-    tarantool> i = s:create_index('primary', {type = 'hash', parts = {{field = 1, type = 'unsigned'}}}
+A recommended design pattern for a data model is to base primary keys on the
+first fields of a tuple, because this speeds up tuple comparison.
 
-The effect is that, for all tuples in space 'tester', field #1 must exist and
-must contain an unsigned integer.
-The index type is HASH, so values in field #1 must be unique, because keys
-in HASH indexes are unique.
+The simple :doc:`SELECT </reference/reference_lua/box_index/select>` request is:
 
-After that, let's define a secondary index (named 'secondary') based on field #2
-of each tuple:
+..  cssclass:: highlight
+..  parsed-literal::
 
-..  code-block:: tarantoolsession
+    :extsamp:`box.space.{**{space-name}**}:select({*{value}*})`
 
-    tarantool> i = s:create_index('secondary', {type = 'tree', parts = {field = 2, type = 'string'}})
+This looks for a single tuple via the first index. Since the first index
+is always unique, the maximum number of returned tuples will be 1.
+You can call ``select()`` without arguments, and it will return all tuples.
+
+An index definition may also include identifiers of tuple fields
+and their expected **types**. See allowed indexed field types in section
+:ref:`Details about indexed field types <details_about_index_field_types>`:
+
+..  cssclass:: highlight
+..  parsed-literal::
 
-The effect is that, for all tuples in space 'tester', field #2 must exist and
-must contain a string.
-The index type is TREE, so values in field #2 must not be unique, because keys
-in TREE indexes may be non-unique.
+    :extsamp:`box.space.{**{space-name}**}:create_index('primary', {type = 'hash', parts = {{field = 1, type = 'unsigned'}}}`
```

Comment:
Below you state "do not use HASH index for primary index". It's quite strange to do it actually.